.travis.yml

```diff
@@ -0,0 +1,106 @@
+language: cpp
+sudo: required
+
+git:
+  depth: 3
+
+env:
+  global:
+    - COMPILE_ASIO=0
+
+matrix:
+  include:
+    # linux with gcc 7
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          install:
+            - gcc-7
+      dist: trusty
+      env:
+        - MATRIX_EVAL="CC=gcc-7 CXX=g++-7"
+      compiler: gcc
+    # linux with gcc 6
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          install:
+            - gcc-6
+      dist: trusty
+      env:
+        - MATRIX_EVAL="CC=gcc-6 CXX=g++-6"
+      compiler: gcc
+    # linux with gcc (default)
+    - os: linux
+      dist: trusty
+      compiler: gcc
+    # linux with clang (default)
+    - os: linux
+      dist: trusty
+      env:
+        - MATRIX_EVAL="COMPILE_ASIO=1"
+      compiler: clang
+    # linux with clang 3.8
+    - os: linux
+      addons:
+        apt:
+          sources:
+            - ubuntu-toolchain-r-test
+          install:
+            - clang-3.8
+      dist: trusty
+      env:
+        - MATRIX_EVAL="CC=clang-3.8 CXX=clang++-3.8 COMPILE_ASIO=1"
+      compiler: clang
+    # osx with xcode9.2/gcc
+    - os: osx
+      osx_image: xcode9.2
+      compiler: gcc
+    # osx with xcode9.2/clang
+    - os: osx
+      osx_image: xcode9.2
+      compiler: clang
+    # osx with xcode/gcc (default)
+    - os: osx
+      compiler: gcc
+    # osx with xcode/clang (default)
+    - os: osx
+      compiler: clang
+
+before_install:
+  - eval "${MATRIX_EVAL}"
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then sudo rm /usr/local/include/c++; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then sudo apt-get -qq update; fi
+
+install:
+  - ci/install-deps.sh
+
+before_script:
+  - ci/build-deps.sh
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then export PKG_CONFIG_PATH=/usr/local/lib/pkgconfig; fi
+  - autoreconf -vfi
+
+script:
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then ./configure --enable-app-bundle; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ./configure --prefix='/usr'; fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then make dmg; fi
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make && sudo make install; fi
+
+after_failure:
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then ls -alhv /usr/bin/ |grep -E "(clang|gcc|g\+\+)"; fi
+  - if [[ -f config.log ]]; then cat config.log; fi
+
+branches:
```

Comment:
Is it necessary to limit the branches?

I normally set Travis-CI to only check PRs, so this setting wouldn't matter, right?

Reply:
@mgeier I think not necessarily, but I wanted to activate it for the feature branches, I'm currently working on. Eventually those can be removed I guess (for now they are useful to me in my clone of the SSR). Having the `master` branch in there is probably never a bad choice though and as long as there is no branch called `travis-integration` or `networking-with-osc`, there shouldn't be any effect in the main SSR repository.

The Travis-CI documentation is sadly not very clear about the "Build pushed branches" feature in the settings. However, from what I understand, if disabled, Travis should not build branches automatically (on commit).

Apart from triggering builds with a pull request ("Build pushed pull requests"), I suggest doing weekly or monthly builds for the master branch though, as the Travis-CI vagrant images are changing constantly (and might break the builds in the process).